Add batch delete tags

JoeffreyChaucer opened the pull request:
#### Checklist
All items have to be completed before a PR is merged

- [x] Adhere to [Contribution Guidelines](https://docs.dxatscale.io/about-us/contributing-to-dx-scale)
- [ ] Updates to Decision Records considered?
- [ ] Updates to documentation at [DX@Scale Guide](https://github.com/dxatscale/dxatscale-guide) considered?
- [ ] Tested changes?
- [ ] Unit Tests new and existing passing locally?

Commit:
feat(publish): Add new flag to delete git tag by age and limit minimum
tags to keep

Commit:
fix(cicd): add a retry to npm tag

azlam-abdulsalam committed:
fix(cicd): fix comment indentation

Commit:
fix(cicd): add docker images build to parallel stage

Commit:
fix(build): fix typo on final message

Commit:
feat(deploy):  add retry when packages fail
to install  on a background job

packages fail to install when there is a background job is running such as
picklist getting updated by earlier
installed package. This does a
infitnite retry when it sees the particular
error message.

Note: Only for EN language

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts@20.24.6

Commit:
feat(deploy): add support to increment retry counts

Packages when deployed as bundles could fail if a previous package is doing an operation, this
feture allows one to set retry counts in pool prepare where sfpowerscripts would attempt to retry
packages. As this is most critical in prepare, it is currently available only on prepare, with other
commands doing an auto retry for activities such a backrground jobs

fixes #1233

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@33.21.0
 - @dxatscale/sfpowerscripts@20.25.0

Commit:
fix(poolconfig): fix mising schema value

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@33.21.1
 - @dxatscale/sfpowerscripts@20.25.1

Commit:
fix(apextests): fix apex tests trigger crashing on unlocked packages

Temporary fix for stanadalone command for triggering apex tests in an org from a given package etc.
This functionality was broken as underlying builder determines it to be unlocked and fails as params
were not set corretly. This work around is an override to source for these sitatuations

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@33.21.2
 - @dxatscale/sfpowerscripts@20.25.2

Commit:
chore(manifest): add license field

Commit:
chore(sfp): merge sfp-cli into sfpowerscripts repository

Commit:
chore(sfp): add sfp to lerna config

Commit:
chore(cicd): add latest tag to sfp-cli

michallachowski committed:
fix(prepare): prevent from downloading / creating artifacts that are not used

fixes #1222

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfp-cli@4.0.1
 - @dxatscale/sfpowerscripts@20.25.3

Commit:
chore(sfp): add oclif to package.json

Commit:
fix(sfp): remove broken reference to oclif-dev

Commit:
chore(deps) :  update dep from npm audit

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfp-cli@4.0.2
 - @dxatscale/sfpowerscripts@20.25.4

azlam-abdulsalam commented on packages/core/src/git/GitTagPurgeListv1.ts:
```diff
@@ -0,0 +1,72 @@
+import simpleGit, { SimpleGit } from 'simple-git';
```
Use the Git abstraction in core lib rather than consuming simple-git directly

azlam-abdulsalam commented on packages/core/src/git/GitTagPurgeListv1.ts:
```diff
@@ -0,0 +1,72 @@
+    this._git = simpleGit();
+  }
+
+  public async listGitTags(pkgs: string[]): Promise<string[]> {
```
See whether you can use 
packages/core/src/git/GitTags.ts
and its functions  especially listTagsOnBranch or modify it

azlam-abdulsalam commented on packages/core/src/git/GitTagPurgeListv1.ts:
```diff
@@ -0,0 +1,72 @@
+        console.log(`no matching tag for ${pkg}`);
+      }
+    }
+    console.log(filteredTags)
```
Do not use console. Logs,  use SFPLogger, as it can be turned of etc

azlam-abdulsalam commented on packages/core/src/git/GitTagPurgeListv1.ts:
```diff
@@ -0,0 +1,72 @@
+    return foundTags;
+  }
+
+  private async deleteGitTags(tagsToDelete: string[]): Promise<void> {
```
Move this to GiTags.ts

azlam-abdulsalam reviewed (changes requested):
See the comments

azlam-abdulsalam committed:
chore(licenses): update dependency licenses

[ci skip]

Commit:
fix(logs): add a new line between installing packages

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts.core@33.21.3
 - @dxatscale/sfp-cli@4.0.3
 - @dxatscale/sfpowerscripts@20.25.5

Commit:
fix(deploy): fix typo on promote message

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfp-cli@4.0.4
 - @dxatscale/sfpowerscripts@20.25.6

  feat(publish): Add delete tag

Commit:
fix(license): remove dep-cruiser as license changed to GPL

Commit:
chore(docs): update license info

Commit:
fix(source-tracking): add more info on error messages

provide additional guidance on error message when fetch is applied on a wrong repository

fixes #1251

Commit:
refactor(pool): remove use of deprecated function

azlam-abdulsalam committed:
refactor(apexlink): move apex link as dxatscale module directly linked (#1252)

* refactor(apexlink): move apex link as dxatscale module directly linked

* fix(tests): add missing test run command

Commit:
chore(deps): remove dependency cruiser

Commit:
fix(deploy): surface metadata deployment error messages properly

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/apexlink@0.0.2
 - @dxatscale/sfpowerscripts.core@33.21.4
 - @dxatscale/sfdx-process-wrapper@0.0.2
 - @dxatscale/sfp-cli@4.0.5
 - @dxatscale/sfpowerscripts@20.25.7

Commit:
chore(dep): add dev dependencies

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/apexlink@0.0.3
 - @dxatscale/sfpowerscripts.core@33.21.5
 - @dxatscale/sfdx-process-wrapper@0.0.3
 - @dxatscale/sfp-cli@4.0.6
 - @dxatscale/sfpowerscripts@20.25.8

Commit:
fix(validateagainstorg): remove default value on disablesourcepkgoverride

default value causes the dependency on devhub to be triggered resulting in incorrect options

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfp-cli@4.0.7
 - @dxatscale/sfpowerscripts@20.25.9

Commit:
chore(deps): fix package-lock.json

Commit:
chore(deps): fix missing dependencies

Commit:
chore(npm): add acess info to new packages

azlam-abdulsalam committed:
feat(validate): disable parallel testing using a flag (#1255)

A new option for validate and validateAgainst Org to disable parallel testing across all the
packages. This could be useful for large projects which are yet to be refactored to support running
tests in parallel or during daily validation runs across unlocked packages on a sandbox. Use
--disableparalleltesting as the option

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/apexlink@0.1.0
 - @dxatscale/sfpowerscripts.core@33.22.0
 - @dxatscale/sfdx-process-wrapper@0.1.0
 - @dxatscale/sfp-cli@4.1.0
 - @dxatscale/sfpowerscripts@20.26.0

Commit:
fix(fetch): fix console logs spilling when used with json flag

fixes #1254

Commit:
fix(prepare): add log grouping

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfp-cli@4.1.1
 - @dxatscale/sfpowerscripts@20.26.1

Commit:
chore(docker): remove apex link from docker

Commit:
chore(deps): remove incorrect dependency on sfp-cli

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfp-cli@4.1.2

Commit:
chore(cicd): remove shrinkwrap in repository

Commit:
fix(schema): disable addtional property validation of sfdx-project.json

This will disable additional property validation of sfdx-project.json, allowing other plugins to
utilise sfdx-project.json

fixes #1256

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfp-cli@4.1.3
 - @dxatscale/sfpowerscripts@20.26.2

Commit:
Revert "chore(cicd): remove shrinkwrap in repository"

This reverts commit a99adcd4d035f02034fd62f2fbc51d70824b7ac3.

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfp-cli@4.1.4

Commit:
chore(npm): fix npm shrinkwrap

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfp-cli@4.1.5

Commit:
chore(npm): add shrinkwrap to commit list

Commit:
fix(tests):fix flakines by increasing timeout

azlam-abdulsalam committed:
fix(releaseconfig): do not use git ref to get release config packages in validate



validate shouldn't require the use of release defn generated from git, as validate could introduce
new packages which are not committed to the current release when done locally or the tag for it is
yet to exit, so it should do a simple filter based on the predicates provided.This will fix the
issue as mentioned in 1259

fixes #1259

azlam-abdulsalam committed:
fix(deploy): ensure max retry count is set to 10 (#1263)

* fix(deploy): ensure max retry count is set to 10

deploy doesnt respect max retry count, it is always set at 2 retries, as it was incorrectly set. The
fix sets a  max default of 10 with a break if earlier if the maxRetry is lower

fixes #1262

* Increase time between retries to 30 seconds

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/apexlink@0.1.1
 - @dxatscale/sfpowerscripts.core@33.22.1
 - @dxatscale/sfp-cli@4.1.6
 - @dxatscale/sfpowerscripts@20.26.3

azlam-abdulsalam committed:
feat(deploy): deploy to support release config (#1264)

Add an option for deploy command to support release config, this will allow deploy to filter
packages to a target org based on the release config

fixes #1232

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts@20.27.0

Commit:
chore(release): update release name

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts@20.28.0

Commit:
feat(conventional-commits): enforce conventional commits

azlam-abdulsalam committed:
feat(build): add support for release config to build (#1272)

Add supports for release config to build & quickbuild.
Utilze the includeOnlyPackages as in validate
to filter the packages in build

azlam-abdulsalam committed:
feat(deploy): enable source tracking support for deploy command (#1273)

rebase changes from main

Co-authored-by: azlam-abdulsalam <azlam.abdulsalam@accenture.com>

Commit:
fix(ci): remove conventional commit on ci

Commit:
chore(publish): update versions and publish to npm

 - @dxatscale/sfpowerscripts@20.29.0

chore(publish): update flag and flag description
